packages/xo-server/src/xapi/index.mjs

```diff
@@ -1523,4 +1524,39 @@ export default class Xapi extends XapiBase {
       }
     }
   }
+
+  async getBiosInfo(ref, { cache } = {}) {
+    const biosData = await this.call('host.get_bios_strings', ref)
+
+    const { 'bios-version': currentBiosVersion, 'system-product-name': hostServerName } = biosData
+
+    if (biosData['system-manufacturer']?.toLowerCase() !== '2crsi') {
+      return
+    }
+
+    // this code has race conditions which might lead to multiple fetches in parallel
+    // but it's no big deal
+    if (!cache?.has('servers')) {
+      const response = await fetch(
+        'https://pictures.2cr.si/Images_site_web_Odoo/Pages_produit/VATES-BIOS_BMC_last-version.json'
+      )
+      const json = await response.json()
+      const servers = keyBy(json[0]['2CRSi_Servers'], 'Server_Name')
+
+      cache?.set('servers', servers)
+    }
+
+    const parsedData = cache?.get('servers')
+
+    const serverData = parsedData?.[hostServerName]
+
+    if (serverData === undefined) {
+      return
+    }
+
+    const { 'BIOS-Version': latestBiosVersion, 'BIOS-link': biosLink } = serverData
+    const isUpToDate = currentBiosVersion === latestBiosVersion
```

Comment:
The PR screenshot shows `currentBiosVersion` left-padded with a `0`, unlike `latestBiosVersion`. Is that normal? If it is, maybe we should compare versions with `semver` instead of `===`.